employeeApp-backend/src/main/java/io/nology/employeeAppbackend/employee/EmployeeController.java

```diff
@@ -0,0 +1,113 @@
+package io.nology.employeeAppbackend.employee;
+
+// Logger imports
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+// Spring framework imports
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+// Java util imports
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/employees")
+public class EmployeeController {
+    
+    Logger user = LoggerFactory.getLogger(EmployeeController.class);
+
+// Dependency injection pattern - CRUD methods
+
+    @Autowired
+    private EmployeeService service;
+
+    // Get methods
+    @CrossOrigin
+    @GetMapping
+
+    public ResponseEntity<List<Employee>> getAll() {
+    List<Employee> allEmployees = this.service.getAll();
+    user.info("Employees successfully found");
+    return new ResponseEntity<>(allEmployees, HttpStatus.OK);
+   }
+
+   @CrossOrigin
+   @GetMapping("/{id}")
+   public ResponseEntity<Employee> getById(@PathVariable Long id) {
+    Optional<Employee> employeeId = this.service.getById(id);
+        if (employeeId.isEmpty()) {
+        user.error("Employee " + id + " was not found"); 
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+        user.info("Employee " + id + " found");
+        return new ResponseEntity<>(employeeId.get(), HttpStatus.OK);
+    }
```

Comment:
catch untrakced exceptions
```java
try {
    Optional<Employee> employeeId = this.service.getById(id);
    if (employeeId.isEmpty()) {
        user.error("Employee " + id + " was not found"); 
        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
    }

    user.info("Employee " + id + " found");
    return new ResponseEntity<>(employeeId.get(), HttpStatus.OK);
} catch (Exception e) {
    logger.error("Something went wrong doing something");
    return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR)
}
```

Reply:
added the above to catch untracked exceptions